common/cache/lru.go

```diff
@@ -263,6 +263,13 @@ func (c *lru) Release(key interface{}) {
 		c.pinnedSize -= entry.Size()
 		metrics.CachePinnedUsage.With(c.metricsHandler).Record(float64(c.pinnedSize))
 	}
+	// Entry size might have changed. Recalculate size and evict entries if necessary.
```

Comment:
so size-based cache only works when the pin option is enabled.

Reply:
Do we ever run with pinning disabled?

Reply:
Size-based cache will work in some cases even if pinning is disabled. It works if you don't change the size of the object after getting it from the cache. One example of this is the history events cache. It is already a size based cache.

If the cache is pinned, after changing the size we will eventually get a call to Release(). But if the cache is not pinned, there is no way for the cache to know if the size was changed.

Reply:
yes events cache, namespace cache don't use pinning.

+1 for making it clear in the comment that size option may not work as expected if element size is changing without the cache knowing and pinning not enabled.